pkg/snmp/traps/server.go

```diff
@@ -0,0 +1,170 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2020 Datadog, Inc.
+
+package traps
+
+import (
+	"net"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/soniah/gosnmp"
+)
+
+/*
+* Traps pipeline diagram
+* ----------------------
+*
+* (devices) --[udp:port1]-!-> (Listener 1) \
+*                         !                 (TrapServer) --[chan]--> (Forwarder) --[chan]--> (Rest of the logs pipeline...)
+* (devices) --[udp:portN]-!-> (Listener N) /
+*
+* ! = delimitation between the outside world and the Agent
+ */
+
+// SnmpPacket is the type of packets yielded by server listeners.
+type SnmpPacket struct {
+	Content *gosnmp.SnmpPacket
+	Addr    *net.UDPAddr
+}
+
+// OutputChannel is the type of the server output channel.
+type OutputChannel = chan *SnmpPacket
+
+// TrapServer runs multiple SNMP traps listeners.
+type TrapServer struct {
+	Started bool
+
+	bindHost    string
+	listeners   []TrapListener
+	output      OutputChannel
+	stopTimeout time.Duration
+}
+
+var (
+	outputChannelSize = 100
```

Comment:
Maybe that one could become a (hidden) configuration setting.

Reply:
This could indeed be a nice knob, it's no huge deal if we don't add it yet as it can be easily included in the future, but nice-to-have.

Reply:
Yes, eventually we could easily make this configurable. For now I'll move this with the other constants you suggested creating. :-)